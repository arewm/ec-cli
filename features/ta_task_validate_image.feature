Feature: Verify Conforma Trusted Artifact Tekton Task
  Verify Conforma Tekton Task feature scenarios

  Background:
    Given a cluster running
    Given stub tuf running
    Given stub git daemon running

  Scenario: Golden container image with trusted artifacts
    Given a working namespace
    Given a snapshot artifact with content:
      ```
      {
		    "components": [
			    {
			      "containerImage": "quay.io/hacbs-contract-demo/golden-container@sha256:e76a4ae9dd8a52a0d191fd34ca133af5b4f2609536d32200a4a40a09fdc93a0d"
			    }
		    ]
		  }
      ```
    Given a cluster policy with content:
      ```
      {
        "publicKey": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAERhr8Zj4dZW67zucg8fDr11M4lmRp\nzN6SIcIjkvH39siYg1DkCoa2h2xMUZ10ecbM3/ECqvBV55YwQ2rcIEa7XQ==\n-----END PUBLIC KEY-----",
        "sources": [
          {
            "policy": [
              "git::github.com/enterprise-contract/ec-policies//policy/release?ref=d34eab36b23d43748e451004177ca144296bf323",
              "git::github.com/enterprise-contract/ec-policies//policy/lib?ref=d34eab36b23d43748e451004177ca144296bf323"
            ],
            "config": {
              "include": [
                "slsa_provenance_available"
              ]
            }
          }
        ]
      }
      ```
    When version 0.1 of the task named "verify-conforma-konflux-ta" is run with parameters:
      | SNAPSHOT_FILENAME       | snapshotartifact                                                                                    |
      | SOURCE_DATA_ARTIFACT    | oci:${REGISTRY}/acceptance/snapshotartifact@${BUILD_SNAPSHOT_DIGEST} |
      | POLICY_CONFIGURATION    | ${NAMESPACE}/${POLICY_NAME}                                                                              |
      | STRICT                  | true                                                                                                     |
      | IGNORE_REKOR            | true                                                                                                     |
      | TRUSTED_ARTIFACTS_DEBUG | "true"                                                                                                   |
      | ORAS_OPTIONS            | --plain-http                                                                                             |
    Then the task should succeed
     And the task logs for step "report-json" should match the snapshot
     And the task results should match the snapshot
