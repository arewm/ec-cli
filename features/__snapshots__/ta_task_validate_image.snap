[Golden container image with trusted artifacts:results - 1]
{
  "TEST_OUTPUT": "{\"timestamp\":\"${TIMESTAMP}\",\"namespace\":\"\",\"successes\":5,\"failures\":0,\"warnings\":0,\"result\":\"SUCCESS\"}\n"
}
---

[Golden container image with trusted artifacts:show-config - 1]
{
  "policy": {
    "sources": [
      {
        "policy": [
          "git::github.com/conforma/policy//policy/release?ref=d34eab36b23d43748e451004177ca144296bf323",
          "git::github.com/conforma/policy//policy/lib?ref=d34eab36b23d43748e451004177ca144296bf323"
        ],
        "config": {
          "include": [
            "slsa_provenance_available"
          ]
        }
      }
    ],
    "publicKey": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAERhr8Zj4dZW67zucg8fDr11M4lmRp\nzN6SIcIjkvH39siYg1DkCoa2h2xMUZ10ecbM3/ECqvBV55YwQ2rcIEa7XQ==\n-----END PUBLIC KEY-----"
  },
  "key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAERhr8Zj4dZW67zucg8fDr11M4lmRp\nzN6SIcIjkvH39siYg1DkCoa2h2xMUZ10ecbM3/ECqvBV55YwQ2rcIEa7XQ==\n-----END PUBLIC KEY-----\n",
  "effective-time": "${TIMESTAMP}"
}
---

[Golden container image with trusted artifacts:report-json - 1]
{
  "success": true,
  "components": [
    {
      "name": "",
      "containerImage": "quay.io/hacbs-contract-demo/golden-container@sha256:e76a4ae9dd8a52a0d191fd34ca133af5b4f2609536d32200a4a40a09fdc93a0d",
      "source": {},
      "successes": [
        {
          "msg": "Pass",
          "metadata": {
            "code": "builtin.attestation.signature_check",
            "description": "The attestation signature matches available signing materials.",
            "title": "Attestation signature check passed"
          }
        },
        {
          "msg": "Pass",
          "metadata": {
            "code": "builtin.attestation.syntax_check",
            "description": "The attestation has correct syntax.",
            "title": "Attestation syntax check passed"
          }
        },
        {
          "msg": "Pass",
          "metadata": {
            "code": "builtin.image.signature_check",
            "description": "The image signature matches available signing materials.",
            "title": "Image signature check passed"
          }
        },
        {
          "msg": "Pass",
          "metadata": {
            "code": "slsa_provenance_available.allowed_predicate_types_provided",
            "collections": [
              "minimal",
              "slsa3",
              "redhat",
              "redhat_rpms",
              "policy_data"
            ],
            "description": "Confirm the `allowed_predicate_types` rule data was provided, since it is required by the policy rules in this package.",
            "title": "Allowed predicate types provided"
          }
        },
        {
          "msg": "Pass",
          "metadata": {
            "code": "slsa_provenance_available.attestation_predicate_type_accepted",
            "collections": [
              "minimal",
              "slsa3",
              "redhat",
              "redhat_rpms"
            ],
            "depends_on": [
              "attestation_type.known_attestation_type"
            ],
            "description": "Verify that the predicateType field of the attestation indicates the in-toto SLSA Provenance format was used to attest the PipelineRun.",
            "title": "Expected attestation predicate type found"
          }
        }
      ],
      "success": true,
      "signatures": [
        {
          "keyid": "",
          "sig": "MEUCIFPod1d9HhGt+TEQPG4j+LINjkifCFFOFrE4jbkvexGGAiEAqSp3ROZUsIOwWro6Tv+lRiR7sdMR0U6Crs1ISuQhHtA="
        }
      ],
      "attestations": [
        {
          "type": "https://in-toto.io/Statement/v0.1",
          "predicateType": "https://slsa.dev/provenance/v0.2",
          "predicateBuildType": "tekton.dev/v1beta1/TaskRun",
          "signatures": [
            {
              "keyid": "SHA256:RHajkr+wMEtGfT2CRFrQEhg/8MY2bDLXVg3F8IuI5nE",
              "sig": "MEUCIHFVZeVR59n9UvN1dwF9Lh3Gv8XWLPDPIIJcnQ8e3TtvAiEA0z/5v6ggvmQyQ1EnYTJo9rwxOYuve4th4P/0639orLg="
            }
          ]
        },
        {
          "type": "https://in-toto.io/Statement/v0.1",
          "predicateType": "https://slsa.dev/provenance/v0.2",
          "predicateBuildType": "tekton.dev/v1beta1/PipelineRun",
          "signatures": [
            {
              "keyid": "SHA256:RHajkr+wMEtGfT2CRFrQEhg/8MY2bDLXVg3F8IuI5nE",
              "sig": "MEUCIQClx1zvZGvyRu5gCHiC+oWVZTmWJGQlocSZMnzx/5omZAIgUiLQuMm+USYE+H0PDn/xPSVVQjkhWjDc3fulkxVzlC0="
            }
          ]
        },
        {
          "type": "https://in-toto.io/Statement/v0.1",
          "predicateType": "https://slsa.dev/provenance/v0.2",
          "predicateBuildType": "tekton.dev/v1beta1/PipelineRun",
          "signatures": [
            {
              "keyid": "SHA256:RHajkr+wMEtGfT2CRFrQEhg/8MY2bDLXVg3F8IuI5nE",
              "sig": "MEUCIGS176zN5aoorLQMukjoCkHm7ocu7UhnKXLhzEdsgp4BAiEAviub3Lf4thLmSTU6ZqnEjw02kkrb9LKBBa1t8hVgAM4="
            }
          ]
        }
      ]
    }
  ],
  "key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAERhr8Zj4dZW67zucg8fDr11M4lmRp\nzN6SIcIjkvH39siYg1DkCoa2h2xMUZ10ecbM3/ECqvBV55YwQ2rcIEa7XQ==\n-----END PUBLIC KEY-----\n",
  "policy": {
    "sources": [
      {
        "policy": [
          "git::github.com/conforma/policy//policy/release?ref=d34eab36b23d43748e451004177ca144296bf323",
          "git::github.com/conforma/policy//policy/lib?ref=d34eab36b23d43748e451004177ca144296bf323"
        ],
        "config": {
          "include": [
            "slsa_provenance_available"
          ]
        }
      }
    ],
    "publicKey": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAERhr8Zj4dZW67zucg8fDr11M4lmRp\nzN6SIcIjkvH39siYg1DkCoa2h2xMUZ10ecbM3/ECqvBV55YwQ2rcIEa7XQ==\n-----END PUBLIC KEY-----"
  },
  "ec-version": "${EC_VERSION}",
  "effective-time": "${TIMESTAMP}"
}
---
